pkg/types/powervs/machinepools.go

```diff
@@ -0,0 +1,55 @@
+package powervs
+
+// +kubebuilder:validation:Enum=capped;dedicated;shared
+type procType string
```

Comment:
This must be a public type so that it can be used in other packages.
```suggestion
type ProcType string
```